transport/conn.go

```diff
@@ -437,3 +447,18 @@ func (c *Conn) Close() {
 func (c *Conn) String() string {
 	return fmt.Sprintf("[addr=%s shard=%d]", c.conn.RemoteAddr(), c.shard)
 }
+
+func (c *Conn) registerEvents(e []frame.EventType) (eventHandler, error) {
+	h := c.r.setEventHandler()
```

Comment:
I fail to understand why the functions are so distant in the code?
They should go one after the other setEventHandler being after registerEvents, it's a helper function.

Reply:
I wanted to have **setEventHandler** next to **setHandler**, but moving it makes more sense.